google/cloud/storage/tests/bucket_integration_test.cc

```diff
@@ -559,6 +559,194 @@ TEST_F(BucketIntegrationTest, IamCRUD) {
   client.DeleteBucket(bucket_name);
 }
 
+TEST_F(BucketIntegrationTest, ListFailure) {
+  Client client;
+
+  // Project IDs must end with a letter or number, test with an invalid ID.
+  auto stream = client.ListBucketsForProject("Invalid-project-id-");
+  std::vector<BucketMetadata> results;
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  EXPECT_THROW(try { results.assign(stream.begin(), stream.end()); } catch (
+                   std::runtime_error const& ex) {
+    EXPECT_THAT(ex.what(), HasSubstr("Permanent error in"));
+    throw;
+  },
+               std::runtime_error);
+#else
+  EXPECT_DEATH_IF_SUPPORTED(results.assign(stream.begin(), stream.end()),
+                            "exceptions are disabled");
+#endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+}
+
+TEST_F(BucketIntegrationTest, CreateFailure) {
+  Client client;
+
+  // Try to create an invalid bucket (the name should not contain `_`), the
```

Comment:
This is true for s3 buckets, but not GCS buckets, as mentioned above

Reply:
Fixed.